sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.AI.FormRecognizer.Training
+{
+    /// <summary>
+    /// The client to use to with the Form Recognizer Azure Cognitive Service to train custom models from forms,
+    /// and to extract values from forms using those custom models.  It also supports listing and deleting trained
+    /// models.
+    /// </summary>
+    public class FormTrainingClient
+    {
+        private readonly ClientDiagnostics _diagnostics;
+        private readonly HttpPipeline _pipeline;
+
+        internal readonly ServiceClient ServiceClient;
+
+        internal const string CustomModelsRoute = "/custom/models";
+
+        /// <summary>
+        /// </summary>
+        protected FormTrainingClient()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FormTrainingClient"/>.
+        /// </summary>
+        public FormTrainingClient(Uri endpoint, AzureKeyCredential credential) : this(endpoint, credential, new FormRecognizerClientOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FormTrainingClient"/>.
+        /// </summary>
+        public FormTrainingClient(Uri endpoint, AzureKeyCredential credential, FormRecognizerClientOptions options)
+        {
+            _diagnostics = new ClientDiagnostics(options);
+            _pipeline = HttpPipelineBuilder.Build(options, new AzureKeyCredentialPolicy(credential, Constants.AuthorizationHeader));
+            ServiceClient = new ServiceClient(_diagnostics, _pipeline, endpoint.ToString());
+        }
+
+        #region Training
+
+        /// <summary>
+        /// Trains a model from a collection of custom forms in a blob storage container.
+        /// </summary>
+        /// <param name="trainingFiles">An externally accessible Azure storage blob container Uri.</param>
+        /// <param name="useLabels"></param>
+        /// <param name="filter">Filter to apply to the documents in the source path for training.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A <see cref="TrainingOperation"/> to wait on this long-running operation.  Its <see cref="TrainingOperation"/>.Value upon successful
+        /// completion will contain meta-data about the trained model.</returns>
+        public virtual TrainingOperation StartTraining(Uri trainingFiles, bool useLabels = false, TrainingFileFilter filter = default, CancellationToken cancellationToken = default)
```

Comment:
Just an FYI that @kinelski has taken a different approach in his last PR where he's kept the old method signatures on the new types until he makes them functional.  Since we want to move fast here, I'm not concerned about the divergence, but just wanted to note it in case we need to be aware of it later.